lib/iris/tests/test_netcdf.py

```diff
@@ -910,6 +910,65 @@ def test_std_name_lookup_fail(self):
         self.assertFalse('phenomenon_time' in iris.std_names.STD_NAMES)
 
 
+class TestCMIP6VolcelloLoad(tests.IrisTest):
+    def setUp(self):
+        dataset = nc.Dataset('issue_3367_volcello_test_file.nc', 'w')
```

Comment:
I like the self-generated testfile,
but I think it's worth creating the file in a "standard" temporary file location, rather than current directory.
That works in case you are running tests from a protected directory (like a system install).

If you pre-create a cube (instead of a file) in the setUp then you can use 
`with self.temp_filename('.nc'):`   (that is, `IrisTest.temp_filename`), and then just save it to file in the test function before reading it back.
I guess that's ok if there's only one testcase, but wasteful if there are more, because the file is always destroyed afterward.  
Another way is to explicitly 'enter' and 'exit' the context manager, instead of the 'with', which can then be done in setUp and tearDown, like the present code.
Or, you can just use `tempfile.mkdtemp()` instead.  [Like this case](https://github.com/SciTools/iris/blob/master/lib/iris/tests/integration/fast_load/test_fast_load.py#L57)

Do whatever seems nicer to you.